synapse/tests/test_lib_auth.py

```diff
@@ -26,3 +39,425 @@ def test_auth_runas(self):
         with s_auth.runas('visi@vertex.link'):
             self.eq(s_auth.whoami(), 'visi@vertex.link')
         self.eq(s_auth.whoami(), 'root@localhost')
+
+    def test_auth_tagtree(self):
+
+        tree = s_auth.TagTree()
+
+        tree.add('foo.bar.baz')
+
+        self.false(tree.get('foo'))
+        self.false(tree.get('foo.bar'))
+
+        self.true(tree.get('foo.bar.baz'))
+        self.true(tree.get('foo.bar.baz.faz'))
+
+        tree.add('foo')
+
+        self.true(tree.get('foo'))
+        self.true(tree.get('foo.bar'))
+        self.true(tree.get('foo.bar.baz'))
+        self.true(tree.get('foo.bar.baz.faz'))
+
+        tree.clear()
+        tree.add('foo.bar.baz')
+        self.false(tree.get('foo'))
+        self.true(tree.get('foo.bar.baz'))
+        # Wildcard takes precedence once it is added
+        tree.add('*')
+        self.true(tree.get('foo'))
+        self.true(tree.get('foo.bar.baz.faz'))
+
+        self.raises(s_exc.BadRuleValu, tree.add, 'hehe.haha.*')
+
+    def test_auth_rbac(self):
+
+        with self.getTestDir() as dirn:
+
+            with s_auth.Auth(dirn) as auth:
+
+                # Ensure our default lmdb size is correct
+                self.eq(auth.getConfOpt('lmdb:mapsize'),
+                        s_const.gigabyte)
```

Comment:
Might want to change this to use the valu from iqtest

Reply:
ah yeah